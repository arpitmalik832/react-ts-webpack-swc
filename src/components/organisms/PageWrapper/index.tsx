import { useEffect } from 'react';
import { Outlet, useLocation } from 'react-router-dom';
import { log } from '@arpitmalik832/react-js-rollup-monorepo-library';

function PageWrapper() {
  const location = useLocation();

  useEffect(() => {
    log('Route changed:', location.pathname);
  }, [location]);

  return (
    <div>
      Page Wrapper
      <Outlet />
    </div>
  );
}

export default PageWrapper;
