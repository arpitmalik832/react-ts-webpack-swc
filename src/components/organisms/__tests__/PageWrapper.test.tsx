import { cleanup, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { log } from '@arpitmalik832/react-ts-rollup-library';

import PageWrapper from '../PageWrapper';

// Mock the react-router dependencies
jest.mock('react-router', () => ({
  __esModule: true,
  ...jest.requireActual<object>('react-router'),
  useLocation: () => ({
    pathname: '/test-path',
  }),
  Outlet: () => <div data-testid="outlet">Outlet Content</div>,
}));

// Mock the logging function
jest.mock('@arpitmalik832/react-ts-rollup-library', () => ({
  log: jest.fn(),
}));

describe('PageWrapper unit tests', () => {
  beforeEach(() => {
    // Clear mock calls before each test
    jest.clearAllMocks();
  });

  afterEach(() => {
    cleanup();
  });

  it('renders PageWrapper with correct content', () => {
    render(<PageWrapper />);

    // Check if "Page Wrapper" text is present
    expect(screen.getByText('Page Wrapper')).toBeInTheDocument();

    // Check if Outlet is rendered
    expect(screen.getByTestId('outlet')).toBeInTheDocument();
    expect(screen.getByText('Outlet Content')).toBeInTheDocument();
  });

  it('logs route change on mount', () => {
    render(<PageWrapper />);

    // Verify that log was called with correct arguments
    expect(log).toHaveBeenCalledTimes(1);
    expect(log).toHaveBeenCalledWith('Route changed:', '/test-path');
  });

  it('PageWrapper snapshot test', () => {
    const { container } = render(<PageWrapper />);
    expect(container).toMatchSnapshot();
  });
});
