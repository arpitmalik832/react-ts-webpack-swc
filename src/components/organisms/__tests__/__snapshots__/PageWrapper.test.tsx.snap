// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper unit tests PageWrapper snapshot test 1`] = `
<div>
  <div>
    Page Wrapper
    <div
      data-testid="outlet"
    >
      Outlet Content
    </div>
  </div>
</div>
`;
